src/HomeWork4/HomeWork4.java

```diff
@@ -0,0 +1,180 @@
+package HomeWork4;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class HomeWork4 {
+    public static void main(String[] args) {
+        /*
+        * Task # 18
+        * Найти в строке не только запятые, но и другие знаки препинания. Подсчитать общее их количество.
+        * */
+        System.out.println("Task #18");
+        String text = "Найти в строке не только запятые, но и другие знаки препинания. Подсчитать общее их количество.";
+        int index = 0;
+        char symbol;
+        for (int i = 0;i < text.length(); i++){
+            symbol = text.charAt(i);
+            if (symbol != 0){
+                if (symbol == '.' || symbol == ',' || symbol == '!' || symbol == '?') {
+                    index ++;
+                }
+            }else {
+                System.out.printf("Знаков препинания нет!");
+            }
+        }
+        System.out.println("Есть " + index + " знаков припинания ");
+        System.out.println("");
+        /*
+        * Task #19
+        * Имеется строка с текстом. Подсчитать количество слов в тексте. Желательно учесть, что слова могут разделяться
+        * несколькими пробелами, в начале и конце
+        * текста также могут быть пробелы, но могут и отсутствовать.
+        *
+        * */
+        System.out.println("Task #19");
+        String text19 = "Имеется строка с текстом. Подсчитать количество слов в тексте. Желательно\n" +
+                "учесть, что слова могут разделяться несколькими пробелами, в начале и конце\n" +
+                "текста также могут быть пробелы, но могут и отсутствовать.";
+        int countText19 = 0;
+        String [] strings = text19.split("\\s+");
+        for (int i = 0; i < strings.length - 1; i++){
+            String text19C = strings[i];
+            if (text19C != null){
+                countText19++;
```

Comment:
also intesting moment. if we need get only amount elements we can get int countText = strings.length